hibernate-core/src/main/java/org/hibernate/mapping/DenormalizedTable.java

```diff
@@ -83,6 +84,18 @@ public void createForeignKeys(MetadataBuildingContext context) {
 		}
 	}
 
+	private ForeignKey createDenormalizedForeignKey(ForeignKey includedTableFk) {
+		final ForeignKey deNormalizedForeignKey = new ForeignKey(this);
+		deNormalizedForeignKey.setReferencedEntityName( includedTableFk.getReferencedEntityName() );
+		deNormalizedForeignKey.setKeyDefinition( includedTableFk.getKeyDefinition() );
+		deNormalizedForeignKey.setReferencedTable( includedTableFk.getReferencedTable() );
+		deNormalizedForeignKey.addReferencedColumns( includedTableFk.getReferencedColumns() );
+		for ( Column keyColumn : includedTableFk.getColumns() ) {
+			deNormalizedForeignKey.addColumn( keyColumn );
+		}
+		return deNormalizedForeignKey;
+	}
+
```

Comment:
One final minor nitpick: `deNormalizedForeignKey` -> `denormalizedForeignKey`, since "denormalize" is one word, not two.

You _should_ be able to just amend your commit and then force push. Unless your `main` branch is protected. (It's usual to create a branch for each pull request, instead of creating a pull request on the `main` branch of your fork.)

Reply:
> One final minor nitpick: deNormalizedForeignKey -> denormalizedForeignKey, since "denormalize" is one word, not two.

Or simply `foreignKey` :-)

Reply:
Done. 
I fixed the typo and changed it to "denormalizedForeignKey".